packages/typescript-estree/src/useProgramFromProjectService.ts

```diff
@@ -19,11 +16,15 @@ export function useProgramFromProjectService(
   parseSettings: Readonly<MutableParseSettings>,
   hasFullTypeInformation: boolean,
 ): ASTAndDefiniteProgram | undefined {
-  const filePath = getCanonicalFileName(parseSettings.filePath);
-  log('Opening project service file for: %s', filePath);
+  const filePathAbsolute = absolutify(parseSettings.filePath);
```

Comment:
```suggestion
  // we don't use canonicalized filename here because it causes massive perf regressions for some reason
  const filePathAbsolute = absolutify(parseSettings.filePath);
```

Or something